src/main/java/jenkins/plugins/gerrit/workflow/GerritCommentStep.java

```diff
@@ -63,26 +63,37 @@ protected Execution(@Nonnull StepContext context) throws IOException, Interrupte
 
     @Override
     protected Void run() throws Exception {
+      GerritApiBuilder gerritApiBuilder = new GerritApiBuilder().stepContext(getContext());
+      GerritApi gerritApi = gerritApiBuilder.build();
+      if (gerritApi == null) {
+        return null;
+      }
+
+      /*
+       * NOTICE:
+       * This may be incorrect as anonymous user may have permissions to comment.
+       */
+      if (gerritApiBuilder.isAnonymous()) {
+        listener.getLogger().format("Gerrit review disabled anonymous%n");
+        return null;
+      }
```

Comment:
I do not exactly know what anonymous is, and how to reproduce, however, it seems like anonymous should be able to contribute comments and this implementation is incorrect.

Reply:
@alonbl nope, an anonymous user cannot provide any comment. However can call any REST API and query information about any public information on Gerrit.